laghos_rom.cpp

```diff
@@ -375,48 +375,100 @@ ROM_Basis::ROM_Basis(ROM_Options const& input, Vector const& S, MPI_Comm comm_)
+                    gfH1[i] = S[i];
+                }
+                gfH1.GetTrueDofs(X);
+                for (int i=0; i<tH1size; ++i)
+                {
+                    (*initX)(i) = X[i];
+                }
 
-            for (int i=0; i<L2size; ++i)
-            {
-                gfL2[i] = S[2*H1size+i];
+                for (int i=0; i<H1size; ++i)
+                {
+                    gfH1[i] = S[H1size+i];
+                }
+                gfH1.GetTrueDofs(V);
+                for (int i=0; i<tH1size; ++i)
+                {
+                    (*initV)(i) = V[i];
+                }
+
+                for (int i=0; i<L2size; ++i)
+                {
+                    gfL2[i] = S[2*H1size+i];
+                }
+                gfL2.GetTrueDofs(E);
+                for (int i=0; i<tL2size; ++i)
+                {
+                    (*initE)(i) = E[i];
+                }
             }
-            gfL2.GetTrueDofs(E);
-            for (int i=0; i<tL2size; ++i)
+            else
             {
-                (*initE)(i) = E[i];
+                initX->read("run/ROMoffset/initX" + std::to_string(input.window));
+                initV->read("run/ROMoffset/initV" + std::to_string(input.window));
+                initE->read("run/ROMoffset/initE" + std::to_string(input.window));
+
+                cout << "Read init vectors X, V, E with norms " << initX->norm() << ", " << initV->norm() << ", " << initE->norm() << endl;
             }
         }
         else
         {
-            initX->read("run/ROMoffset/initX" + std::to_string(input.window));
-            initV->read("run/ROMoffset/initV" + std::to_string(input.window));
-            initE->read("run/ROMoffset/initE" + std::to_string(input.window));
+            if ((input.offsetType == 1 && input.paramOffset && input.window == 0) || input.offsetType == 2)
+            {
+                Vector X, V, E;
+
+                for (int i=0; i<H1size; ++i)
+                {
+                    gfH1[i] = S[i];
+                }
+                gfH1.GetTrueDofs(X);
+                for (int i=0; i<tH1size; ++i)
+                {
+                    (*initX)(i) = X[i];
+                }
+
+                for (int i=0; i<H1size; ++i)
+                {
+                    gfH1[i] = S[H1size+i];
+                }
+                gfH1.GetTrueDofs(V);
+                for (int i=0; i<tH1size; ++i)
+                {
+                    (*initV)(i) = V[i];
+                }
 
-            cout << "Read init vectors X, V, E with norms " << initX->norm() << ", " << initV->norm() << ", " << initE->norm() << endl;
+                for (int i=0; i<L2size; ++i)
+                {
+                    gfL2[i] = S[2*H1size+i];
+                }
+                gfL2.GetTrueDofs(E);
+                for (int i=0; i<tL2size; ++i)
+                {
+                    (*initE)(i) = E[i];
+                }
+
+                initX->write("run/ROMoffset/initX0");
```

Comment:
@siuwuncheung Why does `ROM_Basis` write the offset vectors to file? It seems it would only be necessary for the restore phase, after the online phase. For `offsetType == 2`, the offset is the initial state for this window, so the restore phase would need the offsets written to file for each window, right? If so, why is the window index not in the filename? For `offsetType == 1`, the offset is the initial state at `t=0`, which should have been written to file in the offline phase by `ROM_Sampler`, right?

Reply:
@dylan-copeland Thanks for your review. Yes you're right about the case `offsetType == 2`, in which writing the offset solution is not necessary in the online phase but the restore phase. As you have suggested, the filename should include the window index, which is done by the latest commit. 

For `offsetType == 1`, the offset is written to file in the offline phase by `ROM_Sampler` for the non-parametric case. But for the parametric case, the offset is dependent on the parameter and has to be computed and saved at the first time window. There should be an upcoming fix for the restore phase in `offsetType == 1' too, as it just has one offset file even with multiple time windows.

Reply:
@siuwuncheung Thanks for the update and explanation. It looks like the restore phase is not implemented correctly, as it should read the offset rather than writing it, but are you saying that will be fixed in a future commit or PR?

Reply:
@dylan-copeland Sorry for the confusion caused. What I meant was the changes done in the PR so far was only about the online phase. For `offsetType == 2`, the offset is written to file in the online phase, which is used for future read in the restore phase. The restore phase has not be taken care yet, but shall be done shortly in this PR. On the other hand, for `offsetType == 1`, there needs to be a further modification, as the offset file is only written in the online phase when `input.window == 0`.

Reply:
@siuwuncheung Ok, please let me know when the PR is ready for review.

Reply:
@dylan-copeland The PR is ready for review now, with the restore phase is revised after commit 35f8ba1.